### PR TITLE
ESQL: Fix mv evaluators releasing

### PR DESCRIPTION
**Pull request by nik9000**
This adds comprehensive tests for `ExpressionEvaluator` making sure that it releases `Block`s. It fixes all of the `mv_*` evaluators to make sure they release as well.

### CODE REVIEW

**Comment by nik9000** [comment]
This seems to have plugged in a few more things, exposing more failures.

File: x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/X-MultivalueDedupe.java.st [hunk 1]
```diff
@@ -75,46 +75,48 @@ $endif$
...
-                     * strategies. So we just don't try it for now.
-                     */
-                    if (count < ALWAYS_COPY_MISSING) {
-                        copyMissing(first, count);
-                        writeUniquedWork(builder);
-                    } else {
-                        copyAndSort(first, count);
-                        writeSortedWork(builder);
+        try (ref) {
+            $Type$Block.Builder builder = $Type$Block.newBlockBuilder(block.getPositionCount());
```
Comment (dnhatn) [inline x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/X-MultivalueDedupe.java.st:79]: I think we should build this block from BlockFactory. Let's do this in a follow-up.

**Review by dnhatn — APPROVED** [review]
LGTM. Thanks, Nik!

File: x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/X-MultivalueDedupe.java.st (hunk 1 above)
Comment (nik9000) [reply x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/X-MultivalueDedupe.java.st:79]: ++

**Pull request merged**